packages/react-router/index.tsx

```diff
@@ -884,7 +884,7 @@ export function matchPath(
   let matchedPathname = match[1];
   let values = match.slice(2);
   let params = paramNames.reduce((memo, paramName, index) => {
-    memo[paramName] = safelyDecodeURIComponent(values[index], paramName);
+    memo[paramName] = safelyDecodeURIComponent(values[index] || '', paramName);
```

Comment:
@mjackson Our updated regex below made it to where nothing matches for the final group in the case where the `path` is "react/*" and the pathname is "react". This meant we were getting `undefined` for that group's value, which cause the decoding to fail.

Everything is passing now with this simple fix, but let me know if this is a problem or if I should instead investigate fixing this at the expression building step inside of `compilePath`